### PR TITLE
JBPM-9550 [GSS] (7.8.x) Business Central - Blank process diagram for …

### PR DESCRIPTION
**Pull request by bxf12315**
…processes with long running tasks

**Thank you for submitting this pull request**

**JIRA**:
[JBPM-9550](https://issues.redhat.com/browse/JBPM-9550)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

https://github.com/kiegroup/droolsjbpm-integration/pull/2422

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties [hunk 1]
```diff
@@ -187,3 +187,4 @@ NavigationConfirmOkButtonText=Ok
 NavigationConfirmMessage=Are you sure you want to go to Sub Process Diagram?
 AlertPopupTitle=Can not navigate to Sub Process Diagram
 AlertPopupMessage=There are multiple instances of the sub-process, please use the Sub Process Instances Panel instead.
+CanNotGetTimerInstancesMessage=A long task is running and Timers cannot be displayed. Check again later.
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties:190]: This failure could be caused by other issues so I would recommend living it more generic, something like:
```suggestion
CanNotGetTimerInstancesMessage=Error trying to retrieve process instance timers details. Message: {0}
```

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties:190]: ok

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties:190]: Done

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Review by cristianonicolai — APPROVED** [review]

**Review by nmirasch — APPROVED** [review]

**Comment by prashuts** [comment]
@bxf12315  Followed the steps as per ticket , I see issues after starting 'long.wait' instance - https://issues.redhat.com/browse/JBPM-9575 .
Please have a look once

**Comment by prashuts** [comment]
Jenkins run fdb

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java [hunk 2]
```diff
@@ -162,34 +164,42 @@ protected void loadProcessInstanceDetails() {
...
 
-            view.onShow();
         }).getProcessInstanceDiagramSummary(processInstance.getProcessInstanceKey(),
                                             preferences.getProcessInstanceDiagramCompletedNodeColor(),
                                             preferences.getProcessInstanceDiagramCompletedNodeBorderColor(),
                                             preferences.getProcessInstanceDiagramActiveNodeBorderColor());
     }
 
+    protected void loadTimerInstances() {
+        if (processInstance.getState() == 1) {
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java:181]: do we have the state as a constant somewhere? Would be nice to replace the literal here.

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java]: you are right. Done

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by prashuts** [comment]
@bxf12315 i see issue still persists (https://issues.redhat.com/browse/JBPM-9575) but its random , inconsistent. Please have a look

**Comment by bxf12315** [comment]
> @bxf12315 i see issue still persists ([issues.redhat.com/browse/JBPM-9575](https://issues.redhat.com/browse/JBPM-9575)) but its random , inconsistent. Please have a look

please try again.

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by jstastny-cz** [comment]
@cristianonicolai @nmirasch can you please review again with regards to https://issues.redhat.com/browse/JBPM-9575 and advise?

**Comment by jstastny-cz** [comment]
The Unexpected error dialog is considered a blocking issue, please see how the exception can be caught - I mean we're already processing it using the notifications, so we should have a fair grasp on the exception we're actually processing.

**Comment by cristianonicolai** [comment]
@jstastny-cz the problem is not gone as to access timer info, we actually end up in a timeout sync with the process instance. If a timeout exception is not acceptable we need to look if it is possible to retrieve the same or similar information for timers without ending up in a concurrent call to the engine.

**Comment by jstastny-cz** [comment]
@cristianonicolai I would say we want to handle the exception so that it does not bubble up into logs, nor to UI as Unexpected Error Dialog. I am considering the "Expected result" section of [JBPM-9550)](https://issues.redhat.com/browse/JBPM-9550), which states that no error log appears there (which is tied closely to the Unexpected Error Dialog showing up anyway).

**Comment by cristianonicolai** [comment]
@bxf12315 @jstastny-cz so this is a partial fix only, it doesnt solve the entire issue. It needs to be look into further to try to avoid the exception at all.

**Comment by bxf12315** [comment]
dependent on https://github.com/kiegroup/droolsjbpm-integration/pull/2422

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by prashuts** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Pull request merged**